contrib/polaris/registry.go

```diff
@@ -115,15 +115,16 @@ func (r *Registry) Register(_ context.Context, instance *registry.ServiceInstanc
 		}
 
 		// metadata
-		if instance.Metadata == nil {
-			instance.Metadata = make(map[string]string)
+		rmd := mapClone(instance.Metadata)
```

Comment:
<img width="891" alt="image" src="https://github.com/user-attachments/assets/c9ef47bb-8a5d-480e-a9bc-ca6b9113ac80">
this should also be changed to `rmd`?